aws/session/shared_config.go

```diff
@@ -266,13 +291,18 @@ func (cfg *sharedConfig) setFromIniFiles(profiles map[string]struct{}, profile s
 		// profile only have credential provider options.
 		cfg.clearAssumeRoleOptions()
 	} else {
-		// First time a profile has been seen, It must either be a assume role
-		// credentials, or SSO. Assert if the credential type requires a role ARN,
-		// the ARN is also set, or validate that the SSO configuration is complete.
+		// First time a profile has been seen. Assert if the credential type
+		// requires a role ARN, the ARN is also set
 		if err := cfg.validateCredentialsConfig(profile); err != nil {
 			return err
 		}
 	}
+
+	// if not top level profile and has credentials, return with credentials.
+	if len(profiles) != 0 && cfg.Creds.HasKeys() {
```

Comment:
This block is copied from [go v2](https://github.com/aws/aws-sdk-go-v2/blob/b6ca9e1374b9d9f6ae4cf80be79d8d5866050d77/config/shared_config.go#LL849C14-L849C14), it will be triggered when a source profile can get accessKey/secretAccessKey from shared config files, then the source shared config will be returned with credentials. But there might be some diff between v1 and v2's sharedConfig setting.

Reply:
I thought this part was added to v2 for [resolving credentials](https://github.com/aws/aws-sdk-go/blob/d4aaf18d8d7042787830094d94f4018afe81f165/aws/session/credentials.go#L110) as well.

Reply:
i still dont understand why this is needed in order to parse sso sessions? i dont think we should be copying anything from V2 just because its there that were not sure we also need.

Reply:
I agree, the remaining logic can still load the sso session for shared config. I will delete it.